clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -169,6 +191,24 @@ CompletableFuture<Void> sendAutoCommit(final Map<TopicPartition, OffsetAndMetada
                 });
     }
 
+    /**
+     * @return True if auto-commit is enabled as defined in the config {@link ConsumerConfig#ENABLE_AUTO_COMMIT_CONFIG}
+     */
+    public boolean autoCommitEnabled() {
+        return autoCommitState.isPresent();
+    }
+
+    /**
+     * Reset the auto-commit timer so that the next auto-commit is sent out on the interval
+     * starting from now. If auto-commit is not enabled this will perform no action.
+     */
+    public void resetAutoCommitTimer() {
+        if (!autoCommitState.isPresent()) {
+            return;
+        }
+        autoCommitState.get().resetTimer();
```

Comment:
Nit: more idiomatic:

```suggestion
        autoCommitState.ifPresent(AutoCommitState::resetTimer);
```